### PR TITLE
fix: fix nft order

### PR DESCRIPTION
**Pull request by sahar-fehri**
## Explanation

Fixes the order of nfts on a send flow and when adding Nfts to the account.

### Before
these github issues explain the order issues:

https://github.com/MetaMask/metamask-extension/issues/19875

https://github.com/MetaMask/metamask-extension/issues/19876

### After

image

image

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/helpers/utils/util.js [hunk 1]
```diff
@@ -631,3 +631,19 @@ export const getNetworkNameFromProviderType = (providerName) => {
...
 };
+
+/**
+ * Groups by key an array of objects
+ *
+ * @param {*} key
+ * @param {*} array
+ * @returns array grouped by a key
+ */
+export const groupbyKey = (key, array) => {
```
Comment (darkwing) [inline ui/helpers/utils/util.js:642]: Do we need this or can we use `lodash` `groupBy`?

File: ui/helpers/utils/util.test.js [hunk 2]
```diff
@@ -929,4 +929,53 @@ describe('util', () => {
       expect(util.getNetworkNameFromProviderType('rpc')).toStrictEqual('');
     });
   });
+
+  describe('groupByKey()', () => {
+    it('should group by key an array of objects', () => {
```
Comment (darkwing) [inline ui/helpers/utils/util.test.js:934]: Thank you for tests!  💯

File: ui/pages/send/send-content/send-asset-row/send-asset-row.component.js [hunk 3]
```diff
@@ -69,9 +70,22 @@ export default class SendAssetRow extends Component {
...
+      if (Object.prototype.hasOwnProperty.call(groupedNfts, key)) {
+        groupedNfts[key].sort((a, b) => a.tokenId - b.tokenId);
+      }
+    }
+    const sendableNfts = [];
+    for (const key in groupedNfts) {
+      if (Object.prototype.hasOwnProperty.call(groupedNfts, key)) {
+        sendableNfts.push(...groupedNfts[key]);
+      }
+    }
```
Comment (darkwing) [inline ui/pages/send/send-content/send-asset-row/send-asset-row.component.js:88]: Does `lodash` possibly have a helper that could turn this into one operation?

**Review by darkwing — CHANGES_REQUESTED** [review]
Great start @sahar-fehri !  Just a few questions!

File: ui/helpers/utils/util.js (hunk 1 above)
Comment (sahar-fehri) [reply ui/helpers/utils/util.js]: Definitely! will remove the fct and its tests and replacing it with lodash! 🙏

File: ui/helpers/utils/util.test.js (hunk 2 above)
Comment (sahar-fehri) [reply ui/helpers/utils/util.test.js]: 🙏

File: ui/pages/send/send-content/send-asset-row/send-asset-row.component.js [hunk 4]
```diff
@@ -69,9 +70,19 @@ export default class SendAssetRow extends Component {
 
   async componentDidMount() {
     const sendableTokens = this.props.tokens.filter((token) => !token.isERC721);
-    const sendableNfts = this.props.nfts.filter(
+    const sendableNftsNotSorted = this.props.nfts.filter(
       (nft) => nft.isCurrentlyOwned && nft.standard === TokenStandard.ERC721,
     );
+    // Group and sort the sendableNfts Array
```
[the remark this answers is not in the document]
Comment (sahar-fehri) [reply ui/pages/send/send-content/send-asset-row/send-asset-row.component.js:76]: Thanks, nice one! just pushed! 🙏

**Review by darkwing — APPROVED** [review]
Excellent!

**Review by DDDDDanica — APPROVED** [review]

**Pull request merged**